lib/mdw_fused_dslash.cu

```diff
@@ -597,38 +620,25 @@ namespace quda
       {
         strcpy(aux, meta.AuxString());
         if (arg.dagger) strcat(aux, ",Dagger");
-        //        if (arg.xpay) strcat(aux,",xpay");
         char config[512];
         switch (arg.type) {
-        case MdwfFusedDslashType::D4_D5INV_D5PRE:
-          sprintf(config, ",f0,shift%d,%d,%d,%d,halo%d,%d,%d,%d", arg.shift[0], arg.shift[1], arg.shift[2],
-                  arg.shift[3], arg.halo_shift[0], arg.halo_shift[1], arg.halo_shift[2], arg.halo_shift[3]);
-          strcat(aux, config);
-          break;
-        case MdwfFusedDslashType::D4DAG_D5PREDAG_D5INVDAG:
-          sprintf(config, ",f2,shift%d,%d,%d,%d", arg.shift[0], arg.shift[1], arg.shift[2], arg.shift[3]);
-          strcat(aux, config);
-          break;
-        case MdwfFusedDslashType::D4_D5INV_D5INVDAG:
-          sprintf(config, ",f1,shift%d,%d,%d,%d,halo%d,%d,%d,%d", arg.shift[0], arg.shift[1], arg.shift[2],
-                  arg.shift[3], arg.halo_shift[0], arg.halo_shift[1], arg.halo_shift[2], arg.halo_shift[3]);
-          strcat(aux, config);
-          break;
-        case MdwfFusedDslashType::D4DAG_D5PREDAG:
-          sprintf(config, ",f3,shift%d,%d,%d,%d", arg.shift[0], arg.shift[1], arg.shift[2], arg.shift[3]);
-          strcat(aux, config);
-          break;
-        case MdwfFusedDslashType::D5PRE:
-          sprintf(config, ",f4,shift%d,%d,%d,%d", arg.shift[0], arg.shift[1], arg.shift[2], arg.shift[3]);
-          strcat(aux, config);
-          break;
+        case MdwfFusedDslashType::D4_D5INV_D5PRE: sprintf(config, ",f0"); break;
+        case MdwfFusedDslashType::D4DAG_D5PREDAG_D5INVDAG: sprintf(config, ",f2"); break;
+        case MdwfFusedDslashType::D4_D5INV_D5INVDAG: sprintf(config, ",f1"); break;
+        case MdwfFusedDslashType::D4DAG_D5PREDAG: sprintf(config, ",f3"); break;
+        case MdwfFusedDslashType::D5PRE: sprintf(config, ",f4"); break;
         default: errorQuda("Unknown MdwfFusedDslashType");
         }
+        strcat(aux, config);
+        sprintf(config, "shift%d%d%d%d,halo%d%d%d%d,comm%d%d%d%d", arg.shift[0], arg.shift[1], arg.shift[2],
+                arg.shift[3], arg.halo_shift[0], arg.halo_shift[1], arg.halo_shift[2], arg.halo_shift[3], arg.comm[0],
+                arg.comm[1], arg.comm[2], arg.comm[3]);
+        strcat(aux, config);
       }
 
       template <typename T> inline void launch(T *f, const TuneParam &tp, Arg &arg, const qudaStream_t &stream)
       {
-        setMaxDynamicSharedBytesPerBlock(f);
+        quda::mobius_tensor_core::setMaxDynamicSharedBytesPerBlock(f);
```

Comment:
Why the need for the custom setMaxDynamicSharedBytesPerBlock here?

Reply:
It's because in order to use CUB to do the reduction the kernel will use static shared memory, and for that reason when setting function attributes.
```
qudaFuncSetAttribute((const void *)func, cudaFuncAttributeMaxDynamicSharedMemorySize,
                           deviceProp.sharedMemPerBlockOptin - attr.sharedSizeBytes);
```
I should probably change the `setMaxDynamicSharedBytesPerBlock` in [here](https://github.com/lattice/quda/blob/f5d1db135824bf6df98cf261fecc6810f4cf41fd/include/tune_quda.h#L202). I also want to change the function to a static member function, if you think that's appropiate.

Reply:
Ah, that makes sense.  Do you know if these additional queries have any appreciable overhead?  (I would hope not, but good to check)

Reply:
I checked on nsys and I am not seeing visible overheads (< 2 us). Although it might not be a good idea to set the function attributes every time the kernel is called, the overhead is negeligible.